packages/transport-dom/direct.ts

```diff
@@ -0,0 +1,68 @@
+import { TransportEvent, TransportMessage, isTransportMessage } from './messages';
+import { ConnectError, createPromiseClient } from '@connectrpc/connect';
+import { errorToJson } from '@connectrpc/connect/protocol-connect';
+import { ChannelHandlerFn } from './adapter';
+import { JsonReadOptions, JsonWriteOptions, ServiceType } from '@bufbuild/protobuf';
+import { ChannelTransportOptions, createChannelTransport } from './create';
+
+/**
+ * This creates a port for a channelTransport to enter a router, when provided
+ * the handler entry function.
+ *
+ * The returned function satisfies the `getPort` parameter required by
+ * `createChannelTransport`.
+ *
+ * @param entry router entry function
+ * @returns a getPort parameter for createChannelTransport
+ */
+
+const directGetPort =
+  (entry: ChannelHandlerFn, jsonOptions?: Partial<JsonReadOptions> & Partial<JsonWriteOptions>) =>
+  (): Promise<MessagePort> => {
+    const { port1: servicePort, port2: clientPort } = new MessageChannel();
+
+    // straight-pipe connection, no conditions or transformations
+    const directEntryHandler = async ({ requestId, message }: TransportMessage) => {
+      const transportResponse: TransportEvent = await entry(message)
+        .then(response =>
```

Comment:
Can we choose .then() or awaits here?

Reply:
we actually can't, without getting significantly more complicated. i can collapse the 'catch' into a second 'then' parameter.